rfd/0231-new-oracle-join-method.md

```diff
@@ -0,0 +1,309 @@
+why the policy is necessary.
+
+Configuration on the Teleport side will not change.
+The new design will continue to leverage existing Oracle join tokens that
+Teleport admins may have configured.
+No configuration options will need to change on the agent or bot.
+
+### Design
+
+The new method will leverage OCI instance identity certificates, which are made
+available to all OCI compute instances via the IMDS (Instance Metadata Service).
+The best docs available on this certificate are the FAQ entries
+[here](https://docs.oracle.com/en-us/iaas/Content/Identity/Tasks/callingservicesfrominstances.htm#faq).
+This certificate can be retrieved on any instance with the following `curl` command:
+
+```
+curl -H 'Authorization: Bearer Oracle' http://169.254.169.254/opc/v2/identity/cert.pem
+```
+
+An example instance identity certificate includes the following:
+
+```
+Certificate:
+    Data:
+        Version: 3 (0x2)
+        Serial Number:
+            ca:49:16:9e:30:1f:5a:13:d3:09:2d:11:1b:fb:38:a4
+        Signature Algorithm: sha256WithRSAEncryption
+        Issuer: OU=opc-device:36:9f:ed:ff:cc:b9:a4:a1:ea:71:76:ae:6a:05:d3:fb:c3:70:2a:9c:d9:77:05:ed:d3:3c:d5:50:f7:64:04:e0, CN=PKISVC Identity Intermediate r2
+        Validity
+            Not Before: Oct 24 21:39:42 2025 GMT
+            Not After : Oct 24 23:40:42 2025 GMT
+        Subject: CN=ocid1.instance.oc1.phx.anyhqljrtygmvrycfx7frgsbzaiggh7s7yyq6bokgajeoipfvfrsq456cuna, OU=opc-certtype:instance, OU=opc-compartment:ocid1.compartment.oc1..aaaaaaaaeajccvkj5gws3kd32uhr5zvu2ag6fbxevjlny4hv3k6gum2cicha, OU=opc-instance:ocid1.instance.oc1.phx.anyhqljrtygmvrycfx7frgsbzaiggh7s7yyq6bokgajeoipfvfrsq456cuna, OU=opc-tenant:ocid1.tenancy.oc1..aaaaaaaaopdsxc4qa4natvlx6ns34d56qzz2pyi3is32wxwnwx366ebwdueq
+```
+
+The subject OU fields `opc-instance`, `opc-compartment`, and `opc-tenant`
+describe the instance's ID, compartment, and tenancy.
+
+In this design the instance provides the identity certificate, a chain of
+trust for the certificate, and a challenge signature to the Teleport cluster.
+After the Teleport cluster verifies the certificate was legitimately issued by
+Oracle and the instance proves ownership of the private key corresponding to
+the certificate, Teleport will allow the instance to join the cluster if the
+identity matches an allow rule in a configured Teleport provision token.
+
+#### Verifying the certificate chain
+
+All instance identity certificates are issued and signed by an intermediate
+CA, which is itself issued by one of Oracle's Instance
+Identity Root authorities.
+
+The instance cert and the intermediate CA chain are themselves available via
+the instance IMDS at the following URLs:
+
+```
+http://169.254.169.254/opc/v2/identity/cert.pem
+http://169.254.169.254/opc/v2/identity/intermediate.pem
+```
+
+The joining instance will fetch these and include them in the join request.
+
+However, the Instance Identity Root CAs are only available via an authenticated endpoint:
```

Comment:
It's a shame this endpoint requires authentication and hence still requires us to leverage the request signing flow. Have we had the opportunity to discuss this with Oracle at all? I can't really see any good workarounds though.

Reply:
Fair enough - I suppose we can hope they'll change this eventually. I did wonder about just embedding these roots into Teleport, but, I figure that they are unlikely to give good notification of when they will rotate these roots - so probably not worth the risk 😭

Reply:
The expiry on the root CAs in us-phoenix-1 is Apr 18 2028, in ca-montreal-1 it's Jun 3 2029 (just 2 regions i randomly checked). The JSON response at this endpoint includes a refreshIn field that always seems to be the next UTC midnight (currently it's 8 hours from now `"refreshIn" : "2025-10-31T00:00:00.000Z"`). All this to say I agree it's not worth the risk embedding the certs in Teleport